test/onnx/gen_onnx.py

```diff
@@ -626,6 +627,46 @@ def constant_scalar_test():
     return ([node], [], [y])
 
 
+@onnx_test
+def constant_scalar_test2():
+    x = np.array([]).astype(np.int64)
+    y = helper.make_tensor_value_info('0', TensorProto.INT64, [0])
+
+    node = onnx.helper.make_node(
+        'Constant',
+        inputs=[],
+        outputs=['0'],
+        value=onnx.helper.make_tensor(
+            name='empty_tensor',
+            data_type=TensorProto.INT64,
+            dims=x.shape,
+            vals=x.flatten().astype(np.int64),
+        ),
+    )
+
+    return ([node], [], [y])
+
+
+@onnx_test
+def constant_scalar_test3():
```

Comment:
This test does not generate a scalar but instead a constant value with 1 element. Can you please rename this?

Reply:
Done

Reply:
`constant_scalar_test` is also not generating scalar in that case.